core/modules/replica/MessengerConnector.cc

```diff
@@ -0,0 +1,640 @@
+
+                    request2notify = _currentRequest;
+
+                    _id2request.erase(_currentRequest->id);
+                    _currentRequest = nullptr;
+
+                    restart(lock);
+
+                } else {
+
+                    size_t bytes;
+                    if (syncReadFrame(lock,
+                                      _inBuffer,
+                                      bytes)) {
+
+                        // If something bad happened along the line then we should notify a caller about
+                        // a problem with the request, remove the request from all data structures and
+                        // restart the communication.
+
+                        request2notify = _currentRequest;
+
+                        _id2request.erase(_currentRequest->id);
+                        _currentRequest = nullptr;
+
+                        restart(lock);
+
+                    } else {
+
+                        LOGS(_log, LOG_LVL_DEBUG, context() << "responseReceived"
+                             << "  _currentRequest->id=" << _currentRequest->id
+                             << " bytes=" << bytes);
+
+                        // Then receive response body into a buffer inside the wrapper
+
+                        if (syncReadMessageImpl(lock,
+                                                _currentRequest->responseBuffer,
+                                                bytes)) {
+
+                            // If something bad happened along the line then we should notify a caller about
+                            // a problem with the request, remove the request from all data structures and
+                            // restart the communication.
+
+                            request2notify = _currentRequest;
+
+                            _id2request.erase(_currentRequest->id);
+                            _currentRequest = nullptr;
+
+                            restart(lock);
+
+                        } else {
+
+                            // Finally, success!
+
+                            // Notify a caller about the successfull completion of the operation,
+                            // remove this request from all data structures and initiate a new request.
+
+                            request2notify = _currentRequest;
+                            request2notify->success = true;
+
+                            _id2request.erase(_currentRequest->id);
+                            _currentRequest = nullptr;
+
+                            sendRequest(lock);
+                        }
+                    }
+                }
+            }
+
+        } else {
+
+            // The request submission had a chance to finish (successfully or not)
+            // before the cancellation attempt was made. So, we just forget about
+            // this request and send no notification to a caller.
+            //
+            if (ec) restart(lock);
+            else    sendRequest(lock);
```

Comment:
nees to be on 4 lines, with braces.

Reply:
after consulting with @fritzm, we think that https://developer.lsst.io/cpp/style.html#single-statement-if-else-for-or-while-statements-must-only-be-written-without-brackets-if-on-one-line says it can also be on one line like so `if (ec) restart(lock) else sendRequest(lock);`

Reply:
Added braces. I don't think the one line option makes a terribly readable code here.